exercicios/para-casa/entregas/lais-barreto/projeto/ContaStandard/ContaStandard.test.js

```diff
@@ -0,0 +1,130 @@
+    const contaStandard = new ContaStandard("1234", "12345", 1000, 4000);
+
+    contaStandard.sacar(100);
+    expect(contaStandard.getSaldo()).toBe(900);
+  });
+
+  test("retorna mensagem de erro ao realizar saque com valor negativo", () => {
+    const contaStandard = new ContaStandard("1234", "12345", 1000, 4000);
+
+    expect(() => contaStandard.sacar(-100)).toThrow(
+      "Valor inválido para saque"
+    );
+    expect(contaStandard.getSaldo()).toBe(1000);
+
+    contaStandard.destruir();
+  });
+
+  test("retorna mensagem de erro ao realizar saque com valor maior que o saldo", () => {
+    const contaStandard = new ContaStandard("1234", "12345", 1000, 3000);
+
+    expect(() => contaStandard.sacar(3000)).toThrow("Saldo insuficiente");
+    expect(contaStandard.getSaldo()).toBe(1000);
+
+    contaStandard.destruir();
+  });
+
+  test("retorna mensagem de erro ao depositar valor não numérico", () => {
+    const contaStandard = new ContaStandard("1234", "12345", 1000, 3000);
+
+    expect(() => contaStandard.depositar("string")).toThrow(
+      "Valor inválido para depósito"
+    );
+    expect(contaStandard.getSaldo()).toBe(1000);
+
+    contaStandard.destruir();
+  });
+
+  test("retornar sucesso ao fazer uma transferência com valor válido, saldo suficiente, dados válidos", () => {
+    const contaEmissor = new ContaStandard("1234", "12345", 100, 1000);
+    const contaReceptor = new ContaStandard("1234", "61478", 600, 3000);
+
+    expect(contaEmissor.transferir(10, "1234", "61478")).toBe(
+      "Transferência realizada com sucesso"
+    );
+    expect(contaEmissor.getSaldo()).toBe(90);
+    expect(contaReceptor.getSaldo()).toBe(610);
+
+    contaEmissor.destruir();
+    contaReceptor.destruir();
+  });
+
+  test("retornar mensagem de erro ao fazer uma transferência com saldo insuficiente", () => {
+    const contaEmissor = new ContaStandard("1234", "12345", 10, 1000);
+    const contaReceptor = new ContaStandard("1234", "61478", 600, 3000);
+
+    expect(() => contaEmissor.transferir(75, "1234", "61478")).toThrow(
+      "Saldo insuficiente para transferência."
+    );
+
+    contaEmissor.destruir();
+    contaReceptor.destruir();
+  });
+
+  test("retornar mensagem de erro ao fazer uma transferência para uma conta inválida", () => {
+    const contaEmissor = new ContaStandard("1234", "12345", 10, 1000);
+
+    expect(() => contaEmissor.transferir(500, "2222", "12514")).toThrow(
+      "Conta bancária não encontrada"
+    );
+
+    contaEmissor.destruir();
+  });
+
+  test("retornar mensagem de erro ao fazer uma transferência com valor inválido", () => {
+    const contaEmissor = new ContaStandard("1234", "12345", 10, 1000);
+    const contaReceptor = new ContaStandard("1234", "61478", 600, 3000);
+
+    expect(() => contaEmissor.transferir(-30, "1234", "61478")).toThrow(
+      "Valor inválido para transferência"
+    );
+
+    contaEmissor.destruir();
+    contaReceptor.destruir();
+  });
+});
```

Comment:
**Testes**
* Alterar a descrição dos testes usando práticas da convenção como (Should ou Deve). Ex: test("Deve verificar se instancia Cliente está sendo criada"


**class Cliente**
* Alterar o nome do método registrar para registrarCliente deixando mais descritivo o objetivo do método.

**class Conta**
* Dividir em métodos menores para que cada método faça apenas uma coisa.
	- Em criarConta, dividir a validação da conta em um outro método.
	- Em sacar  e depositar, dividir a validação do valor em outro método, para que possa ser usado nos dois métodos sem a necessidade de repetição de código.
	- Em transferir e  transferirViaPix  é possível criar um método para fazer as validações e usar esse método para confirmar a transferência.
	- Também é possível criar métodos separados que setam os valores das contas e então  fazer a chamada deles após a confirmação da transferência.
	- Em criarChavePix, seria legal criar palavras mágicas para os tipos de chaves  e para os tipos de regex e usar na construção do código para que seja mais legível e fácil de entender o que ele faz.
Ex: 
const TIPO_DE_CHAVE_PIX = {
CPF,
EMAIL,
TELEFONE

**class {ContaStandart, ContaGold, ContaPremium}**
* Em criarConta fazer a verificação apenas da renda, já que a verificação da conta e saldo foi realizada na classe mãe.
* Em transferi e transferirPix criar um método separado para verificação de limite transacional.

**class ContaPremium**
* Retirar a verificação de limite transacional, pois esse tipo de conta não possui limite.